WNPRC_EHR/src/org/labkey/wnprc_ehr/pages/husbandry/WaterCalendar.jsp

```diff
@@ -0,0 +1,1540 @@
+    List<JSONObject> assignedToList= JsonUtils.getSortedListFromJSONArray(assignedToOptions.getResults().getJSONArray("rows"),"title");
+
+    SimpleQuery husbandryFrequency = queryFactory.makeQuery("ehr_lookups", "husbandry_frequency");
+    List<JSONObject> husbandryFrequencyList= JsonUtils.getSortedListFromJSONArray(husbandryFrequency.getResults().getJSONArray("rows"),"altmeaning");
+
+    SimpleQuery husbandryFruit = queryFactory.makeQuery("ehr_lookups", "husbandry_fruit");
+    List<JSONObject> husbandryFruitList= JsonUtils.getSortedListFromJSONArray(husbandryFruit.getResults().getJSONArray("rows"),"title");
+
+    //TODO: Query WaterCoalesce for all future water for the next 60 days
+    SimpleQuery futureWaters = queryFactory.makeQuery("study", "waterScheduleCoalesced");
+    List<JSONObject> waterList = JsonUtils.getListFromJSONArray(futureWaters.getResults().getJSONArray("rows"));
+
+    //SimpleQuery waterAccessControlled = queryFactory.selectRows("wnprc", "watermonitoring_access");
+    List<JSONObject> waterAccess = JsonUtils.getListFromJSONArray(queryFactory.selectRows("wnprc", "watermonitoring_access"));
+
+    JSONObject userAccessWater = new JSONObject();
+
+    for(JSONObject json : waterAccess){
+        CaseInsensitiveHashMap<String> map = new CaseInsensitiveHashMap(json);
+
+        String allowUser = String.valueOf(map.get("alloweduser"));
+        if (userAccessWater.isNull(allowUser)){
+            if (map.get("project") != null){
+                JSONArray projectList = new JSONArray();
+                projectList.put(map.get("project"));
+                userAccessWater.put(allowUser, projectList);
+            }
+        }else {
+            if (map.get("project") != null)
+            {
+                JSONArray projectList = userAccessWater.getJSONArray(allowUser);
+                projectList.put(map.get("project"));
+            }
+
+        }
+
+
+    }
+    String userid = String.valueOf(getUser().getUserId());
+    boolean isAdmin =  getUser().isInSiteAdminGroup();
+
+    JSONObject husbandryAssignmentLookup = new JSONObject();
+    List<JSONObject> husbandryAssigned = JsonUtils.getListFromJSONArray(queryFactory.selectRows("ehr_lookups", "husbandry_assigned"));
+
+    for (JSONObject json : husbandryAssigned) {
+        CaseInsensitiveHashMap<String> map = new CaseInsensitiveHashMap(json);
+        JSONObject waterInfo = new JSONObject();
+
+        if (map.get("category") != null) {
+            waterInfo.put("color", map.get("category"));
+        } else {
+            waterInfo.put("color", "#F78181");
+        }
+
+        String title = map.get("title");
+        waterInfo.put("title", (title != null) ? title : map.get("value"));
+
+        husbandryAssignmentLookup.put(map.get("value"), waterInfo);
+    }
+
+    Group vetGroup = GroupManager.getGroup(getContainer(), "veterinarians (LDAP)", GroupEnumType.SITE);
```

Comment:
Might be good to use labkey roles here instead of hard coding groups

Reply:
We should add this as a TODO